### DISCUSSION TITLE
SMTP Server Access

### DISCUSSION THREAD

**Discussion by pkail** [vercel/vercel#4857] [Help] [answered]
Vercel is a joy to use...except that we can't use it, because it blocks SMTP server requests.  Our applications have a contact form which sends an email to the client (not an unusual setup).  At present, our only solution is to move everything to Digital Ocean, or, alternatively, to rewrite the back-end using Express and host it separately.

Is there any chance this will be fixed in the future?

**Comment by styfle** [comment, marked as answer]
No plans to change the [SMTP limitation](https://vercel.com/docs/v2/platform/limits#smtp).

Checkout [this article](https://vercel.com/knowledge/serverless-functions-and-smtp) for some solutions.

I've also had success using a REST API via mailgun.com to send an email from a Serverless Function.

**Reply by janzheng** [reply]
I'm using nodemailer with mailgun.com to send emails but maybe one in every four emails gets a "connection timeout" and I'm sending like... 1 email every hour, so it's not like I'm sending lots. Everything seems to work locally, but Vercel seems super spotty for some reason. Have you seen this problem before?

**Reply by pkail** [reply]
My guess is that whoever is doing the blocking isn't doing it very well.

**Reply by janzheng** [reply]
haha well, we're supposed to be able to use nodemailer with mailgun — that's even the official solution. But yet, somehow the connection keeps timing out?

**Reply by janzheng** [reply]
OH hmm ok got it, they're supposed to be blocking nodemailer haha

**Reply by janzheng** [reply]
ok so I guess I just switched to using `https://www.npmjs.com/package/mailgun.js` and instead of using SMTP I'm using the API key... which for some reason Mailgun can't generate one API key per SMTP user or even domain — it uses on API key for the ENTIRE account, which seems super silly to me and will probably cause more email spam havoc, but that's life I guess!

**Comment by pixelprodev** [comment]
I'm a bit confused here - I have multiple projects successfully sending smtp emails out using nodemailer at the moment.  What exactly is blocked and do I have any reason to believe these apps (being used by clients in production) are suddenly going to stop sending emails?

The nodemailer code is connecting using `smtpTransport` to my mail accounts (one private hosted and one Amazon SES) over using TLS on port 587 - is that why mine is working? I notice in your article linked it recommends Amazon SES as one solution - does that mean this approach is considered secure and not blocked by your platform?

I'm just trying to understand if I'm going against the grain or if the statement "we block SMTP traffic" needs to be clarified a bit more.

**Reply by styfle** [reply]
Hi @pixelprodev 

Most of Vercel's Limitations (including SMTP) is a limitation of our upstream providers, not something that we explicitly block. We just want to make it clear for our users upfront to avoid confusion if/when SMTP does get blocked.

I'll see if we can get more details about why Amazon SES works, thanks!

**Reply by styfle** [reply]
I updated the article to clarify REST API https://github.com/vercel/docs/pull/1998

**Reply by icepaq** [reply]
Hi Styfle. The PR that you linked isn't a valid link any more.

I'm thinking of writing my own API to send emails and host it on a VPS. 

How do Vercel users usually get around this limitation?

**Comment by pkail** [comment]
>>How do Vercel users usually get around this limitation?
They move to another platform.
